DEVELOPMENT.md

```diff
@@ -0,0 +1,107 @@
+
+The details of the required dependencies will not be documented here, so it is recommended to just let the scripts handle it - you can see what will be installed in `scripts/install_dependencies.sh`
+
+Build:
+
+```
+bash build.sh <arch>
+```
+
+Clean:
+
+```
+bash clean.sh <arch>
+```
+
+Current architectures:
+* linux_x86_64
+* windows_amd64
+* darwin
+
+Final packages will be deployed in the `./_packages/build_<arch>/` directory.
+
+By default the scripts will not build nextpnr-ecp5. See [this section](#building-nextpnr-ecp5) for details on how to build this too.
+
+## Disabling parts of the build
+
+The build scripts define many variables that may be used to disable parts of the build during development - see `build.sh` for details.
+
+If you place a `.env` file in the root of the repo then the bash scripts will source it. You can use this as a convenient way to locally override these variables without accidentally committing changes (the file is already in `.gitignore`).
+
+## Building nextpnr-ecp5
+
+These scripts currently require an Ubuntu 20.04 environment ([as specified here](#prerequisites)) to generate the ECP5 device databases. The device databases are represented as a text-based set of instructions for the nextpnr Binary Blob Assembler (a "BBA file"). It is important that the same nextpnr and libtrellis git commits are used for the whole build to avoid the BBA files getting out of sync with the compiled code.
+
+In the Ubuntu 20.04 environment, run:
+
+`./build_bba.sh`
+
+This will result in a package generated at:
+
+`./_packages/build_linux_x86_64/ecp5-bba-noarch-nightly.tar.gz`
+
+If you are building in the same working folder, you can then simply run:
+
+`COMPILE_NEXTPNR_ECP5=1 ./build.sh <arch>`
+
+If you are building the bba files in a separate working folder, you will need to copy the bba package into `./_packages/build_linux_x86_64/ecp5-bba-noarch-nightly.tar.gz` before running `build.sh` (this is how the CI build works)
+
+Some other info that may be useful if trying to build nextpnr on a new platform:
+
+ * The text BBA files are architecture independent.
+ * The binary output from bbasm works for any architecture provided that the correct endianness was set. This is worth paying attention to if e.g. cross-building for a big-endian platform on an x86 host
+ * The BBA files cannot be generated without a version of libtrellis built with python bindings enabled (they are used by a python script).
+ * Nextpnr-ecp5 links against libtrellis but does not need the python bindings.
+ * Nextpnr is linked against a static libpython.a to enable an embedded python interpreter that may be used to set up clock constraints, manually place elements, etc. This embedded interpreter needs the modules in the `lib/python3.<x>` to function. It is far from perfect - many modules that have been copied over are likely to fail to load since they have dependencies on shared libraries from the build host that we have not bundled.
```

Comment:
Nice. The exact use of python on the target architecture is good to know. Python is what [gave me an headache](https://github.com/open-tool-forge/fpga-toolchain/issues/46#issuecomment-704568477) when trying to do the ARM64 cross-build from Ubuntu.